py3status/modules/external_script.py

```diff
@@ -51,11 +57,13 @@ class Py3status:
     # available configuration parameters
     cache_timeout = 15
     format = '{output}'
+    format_notification = None
     localize = True
     script_path = None
     strip_output = False
 
     def post_config_hook(self):
+        self.button_refresh = 2
```

Comment:
Okay because without `on_click`, the module still does not refresh on any other click events than `2` and we don't have any other button, so we don't need to mention it... and the behavior still will be same as before, but with optional notifications on click events (except `2`) this time.

`button_refresh` is not needed and `{output_full}` is not documented, but this is implemented to do what you want... and `having format_notification` implemented correctly is what I want.

Reply:
why is this feature hidden?  It should either be documented or removed imho

Reply:
New `{output_full}` is the true `{output}`.
... and `{output}` is the true `{line_1}` since it does `output[0]` in the code.

We keep `{output_full}` output as-is... for notifications, but will remove newlines for the bar (although undocumented).

And we should address #1265 PR too because I assume `async_script` could want `{output_full}` too... so maybe we should address all of this in one swift PR... or something like that.

Most users are using `{output}` so we can't swap that. If we have to break something, I prefer we do it once instead of twice, thrice.

Reply:
For this one shift PR, you already have a prototype for rewritten external_script, right?

Reply:
I did start something particularly to deal with explicit lines and to parse py3status's format in the output lines too instead of external_script's second line hidden color... I don't have `Popen` in there, and I pretty much stopped working on that already. See what @tobes want to do here.

@tobes He need `{output_full}` for his notifications and `format_notifications` are made to work with `last` (data), not `current` data like most modules... so I wanted to add `notification` (state) where users can specify what notification state to use (current, last) (default: current).